### PR TITLE
Add `renv_install_local()`

### LINKED ISSUE #11: Provide shortcut for renv::install(pkgbuild::build(...))
```r
dir.create("renv.local", ...)
renv::install(pkgbuild::build("../other.package", dest = "renv/local"))
```

Substitute `renv.local`  with the correct path queried from renv.

Could also live in renv, but would have to copy relevant parts of pkgbuild.

### PR DESCRIPTION
**Pull request by pat-s**
fixes #11 

Please ignore the `lintr` and other artifacts 🐟

### CODE REVIEW

File: R/renv.R [hunk 1]
```diff
@@ -282,3 +282,42 @@ renv_switch_r_version <- function(version = NULL
...
+#' 3. Installing the package from the cache location via `renv::install()`.
+#' @param path `[character]`\cr
+#'   The path to the package which should be built and installed.
+#' @importFrom renv install
+#' @export
+renv_install_local <- function(path = ".") {
+  if (path == ".") {
+    path <- usethis::proj_get()
+  }
+  if (Sys.getenv("RENV_PATHS_LOCAL") == "") {
```
Comment (krlmlr) [inline R/renv.R:305]:
```suggestion
  if (renv_local == "") {
```

and move the assignment up?

File: R/renv.R [hunk 2]
```diff
@@ -282,3 +282,42 @@ renv_switch_r_version <- function(version = NULL
...
+  } else {
+    renv_local <- Sys.getenv("RENV_PATHS_LOCAL")
+  }
+  dir.create(renv_local, showWarnings = FALSE, recursive = TRUE)
+
+  pkg_name <- desc::desc_get_field("Package")
+
+  cli::cli_alert_info("Building package {.field {pkg_name}} and installing into
+    {.field {renv_local}}.", wrap = TRUE)
+  pkg_source <- pkgbuild::build(path, dest_path = renv_local, quiet = TRUE)
```
Comment (krlmlr) [inline R/renv.R:320]:
```suggestion
  pkg_source <- pkgbuild::build(path, dest_path = renv_local, ...)
```

and add `..., quiet = TRUE` to the signature of this function?

**Review by krlmlr — COMMENTED** [review]
Nice, thanks!

**Comment by pat-s** [comment]
@krlmlr Can this be merged?

File: R/renv.R [hunk 3]
```diff
@@ -308,3 +308,52 @@ renv_switch_r_version <- function(version = NULL
...
+  }
+
+  renv_local <- Sys.getenv("RENV_PATHS_LOCAL")
+
+  if (renv_local == "") {
+    renv_local <- switch(Sys.info()[["sysname"]],
+      "Darwin" = "~/Library/Application Support/renv",
+      "Windows" = "%LOCALAPPDATA%/renv",
+      "Linux" = "~/.local/share/renv"
+    )
```
Comment (krlmlr) [inline R/renv.R:343]: Maybe:

```suggestion
    renv_local <- renv::paths$root()
```

?

File: tests/testthat/test-renv.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
+test_that("renv_install_local works", {
+  tmp <- tempdir()
+  options(usethis.quiet = TRUE)
+  usethis::create_package(tmp, open = FALSE)
+
+  cap <- capture.output(suppressMessages(renv_install_local(tmp)))
+  rm(cap)
+
+  renv_local <- switch(Sys.info()[["sysname"]],
```
Comment (krlmlr) [inline tests/testthat/test-renv.R:9]: Same?

**Review by krlmlr — COMMENTED** [review]
Looks good, I wonder if we really need to hard-code the paths.

File: R/renv.R (hunk 3 above)
Comment (pat-s) [reply R/renv.R:343]: Nice! 🙏

**Pull request merged**